tests/test_debugging.py

```diff
@@ -52,6 +52,41 @@ def circuit():
         assert all(k1 == k2 for k1, k2 in zip(result.keys(), expected.keys()))
         assert all(np.allclose(v1, v2) for v1, v2 in zip(result.values(), expected.values()))
 
+    @pytest.mark.parametrize("method", [None, "backprop", "parameter-shift", "adjoint"])
+    def test_default_qubit2(self, method):
+        """Test that multiple snapshots are returned correctly on the new
+        state-vector simulator."""
+        dev = qml.devices.experimental.DefaultQubit2()
+
+        # TODO: change this to use QNodes once the new device supports it
```

Comment:
```suggestion
        # TODO: add additional QNode test once the new device supports it
```

We also need unit tests for (at least) `apply_operation` and `simulate`.